earth2studio/data/utils.py

```diff
@@ -69,6 +74,65 @@ def fetch_data(
 
     return prep_data_array(xr.concat(da, "lead_time"), device=device)
 
+def build_dataset(
```

Comment:
Just a heads up, need some unit tests.

I would recommend adding these edits to the "Random" data source and making tests with that... coverage > 92% please :)